widget/embedded/src/components/ConfirmWalletsModal/ConfirmWalletsModal.tsx

```diff
@@ -298,151 +319,138 @@ export function ConfirmWalletsModal(props: PropTypes) {
-                              fullWidth
-                              suffix={
-                                <ExpandedIcon
-                                  orientation={
-                                    showCustomDestination ? 'up' : 'down'
-                                  }>
-                                  <ChevronDownIcon size={10} color="gray" />
-                                </ExpandedIcon>
-                              }>
-                              <div className="button__content">
-                                <WalletIcon size={18} color="info" />
-                                <Divider size={4} direction="horizontal" />
-                                <Typography variant="label" size="medium">
-                                  Send to a different address
-                                </Typography>
+                            }}>
+                            <Trigger
+                              onClick={setShowCustomDestination.bind(
+                                null,
+                                (prevState) => !prevState
+                              )}>
+                              <CustomDestinationButton
+                                fullWidth
+                                suffix={
+                                  <ExpandedIcon
+                                    orientation={
+                                      showCustomDestination ? 'up' : 'down'
+                                    }>
+                                    <ChevronDownIcon size={10} color="gray" />
+                                  </ExpandedIcon>
+                                }>
+                                <div className="button__content">
+                                  <WalletIcon size={18} color="info" />
+                                  <Divider size={4} direction="horizontal" />
+                                  <Typography variant="label" size="medium">
+                                    Send to a different address
+                                  </Typography>
+                                </div>
+                              </CustomDestinationButton>
+                            </Trigger>
+                            <CollapsibleContent open={showCustomDestination}>
+                              <>
+                                <Divider size={4} />
+                                <StyledTextField
+                                  autoFocus
+                                  placeholder="Your destination address"
+                                  value={destination}
+                                  onChange={(e) => {
+                                    setDestination(e.target.value);
+                                  }}
+                                />
+                              </>
+                            </CollapsibleContent>
+                          </CollapsibleRoot>
+                          {!!destination &&
```

Comment:
`const isAddressMatched = !!destination && showCustomDestination && lastStepToBlockchainMeta && !isValidAddress(lastStepToBlockchainMeta,destination)` or something like this.